### PR TITLE
Tools: make some message catch more robust

### PR DESCRIPTION
**Pull request by khancyr**
Some fix and improvement on getting the message on autotest.
I know that we don't do test with multiple systems yet, but those checks are cheap and will help in the future !

### CODE REVIEW

File: Tools/autotest/common.py [hunk 1]
```diff
@@ -3773,6 +3773,9 @@ def add_param_value(mav, m):
                             self.context_get().parameters.append((name, original_values[name]))
                     received.add(name)
                     continue
+                if original_values[name] == autopilot_values[name] and i % 2 == 0:
+                    self.send_get_parameter_direct(name)
+                    self.progress("Requesting (%s) (retry=%u)" % (name, i))
```
Comment (peterbarker) [inline Tools/autotest/common.py:3778]: Why?  Needs a comment.  I'm doubtful this is required.

I don't really want to change the autotest suite to allow for really lossy links if that's what htis is about.

File: Tools/autotest/common.py [hunk 2]
```diff
@@ -4036,6 +4039,10 @@ def run_cmd(self,
                 target_compid=None,
                 timeout=10,
                 quiet=False):
+        if target_sysid is None:
+            target_sysid = self.target_system
```
Comment (peterbarker) [inline Tools/autotest/common.py:4043]: The RHS doesn't exist AFAICS.

File: Tools/autotest/common.py [hunk 3]
```diff
@@ -4063,7 +4070,7 @@ def run_cmd_get_ack(self, command, want_result, timeout, quiet=False):
             m = self.mav.recv_match(type='COMMAND_ACK',
                                     blocking=True,
                                     timeout=0.1)
-            if m is None:
+            if m is None or m.get_srcSystem() != self.target_system:
```
Comment (peterbarker) [inline Tools/autotest/common.py:4073]: This should come in from a parameter.

File: Tools/autotest/common.py [hunk 4]
```diff
@@ -5698,7 +5705,9 @@ def poll_home_position(self, quiet=True, timeout=30):
                 continue
             m = self.mav.messages.get("HOME_POSITION", None)
             if m is None:
-                continue
+                m = self.mav.recv_match(type='HOME_POSITION', blocking=True, timeout=5)
```
Comment (peterbarker) [inline Tools/autotest/common.py:5708]: This function looks to have been carefully constructed NOT to use `recv_match`.

Was the current implementation causing problems?

File: Tools/autotest/common.py (hunk 1 above)
Comment (khancyr) [reply Tools/autotest/common.py:3778]: I am connected on serial2 on SITL and it fails to get  the set response, so I need to explictly ask for a get

File: Tools/autotest/common.py (hunk 3 above)
Comment (khancyr) [reply Tools/autotest/common.py:4073]: Agree

File: Tools/autotest/common.py (hunk 2 above)
Comment (khancyr) [reply Tools/autotest/common.py:4043]: RHS?

**Comment by peterbarker** [comment]
On Sat, 27 Feb 2021, Pierre Kancir wrote:

> In Tools/autotest/common.py:
> 
> > +                if original_values[name] == autopilot_values[name] and i % 2 == 0:
> +                    self.send_get_parameter_direct(name)
> +                    self.progress("Requesting (%s) (retry=%u)" % (name, i))
> 
> I am connected on serial2 on SITL and it fails to get the set response, so I need to explictly ask for a get

OK, I'm less than keen on bending autotest for lossy links.  We *require* 
the links to be non-lossy - thus our use of tcp.

**Pull request closed without merging**